source/style-guide/style/version.txt

```diff
@@ -0,0 +1,94 @@
+.. _version-reference:
+
+==================
+Version Reference
+==================
+
+.. default-domain:: mongodb
+
+Documentation of features often requires reference to a specific or at least
+major release of the product to indicate when the feature became available,
+when the feature was changed, and when it was deprecated.
+
+Readers find this information useful as it can explain unexpected behavior in
+cases where they attempt to perform an action that is not possible on their
+specific release of the product.
+
+With the exception of release notes and changelogs, which serve as a historical
+document, we should not reference releases of products that we no longer support.
+That is, when a release of MongoDB Server reaches EOL,  references to this
+release should be removed.
+
+* Users don't need to know a feature was introduced in a release that is EOL,
+  since it suggests a text that is out of date.
+
+* Users don't need to know about features that were removed in a release 
+  that is EOL, since it means the feature is unavailable in all supported 
+  releases.
+
+* Users don't need to know about features that were changed in a release
+  that is EOL, since the change now applies to all available releases.
+
+This topic provides guidelines on how to reference product versions in order to
+make it easier to find these references once the release reaches EOL.
+
+General Guidelines
+------------------
+
+When referencing a particular release of a product:
+
+* Prefer version directives wherever possible, as this is the easiest target
+  to later find and evaluate:
+
+  .. code-block:: rst
+
+     .. versionadded:: 6.0
+
+  In cases where the reference requires further explanation, add the content 
+  to the directive block:
+
+  .. code-block:: rst
+
+     .. versionchanged:: 4.4
+
+        Starting in MongoDB 4.4, this command returns the following output:
+
+  Snooty supports three version directives: ``versionadded``, 
+  ``versionchanged``, and ``deprecated``
+
+  .. note::
+      
+     Do **not** include multiple version numbers in this directive, as it
+     does not render correctly in HTML.
+
+* In cases where you need to refer to a version inline, include the product
+  name before the version number, (e.g. ``MongoDB 5.0``).  This ensures that
+  a search for version numbers will correctly match product and release for
+  checks.
+
+* Do **not** reference versions freeform or with the phrase ``version X.X``
+  as these are much harder to identify programmatically as references to 
+  a specific product version.
+
+Identifying EOL Releases
+------------------------
+
+`versc <https://github.com/kennethdyer/versc>`__ is a tool for finding references
+to EOL releases of products.  It operates similar to grep, performing a 
```

Comment:
Suggestion:
If I understand the functionality of the tool, I think it could be useful to identify whether these are inline text references or version directive references (or both).

Reply:
I'm in early stages of redesigning versc to match products to numbers and find other cases that might be hiding in the text.  Until that's ready, I've replaced this section with an example using grep.